src/composables/useRaycaster/index.ts

```diff
@@ -84,17 +84,20 @@ export const useRaycaster = (
 
   const onPointerLeave = (event: PointerEvent) => eventHookPointerMove.trigger({ event, intersects: [] })
 
-  canvas.value.addEventListener('pointerup', onPointerUp)
-  canvas.value.addEventListener('pointerdown', onPointerDown)
-  canvas.value.addEventListener('pointermove', onPointerMove)
-  canvas.value.addEventListener('pointerleave', onPointerLeave)
-
-  onUnmounted(() => {
-    if (!canvas?.value) return
-    canvas.value.removeEventListener('pointerup', onPointerUp)
-    canvas.value.removeEventListener('pointerdown', onPointerDown)
-    canvas.value.removeEventListener('pointermove', onPointerMove)
-    canvas.value.removeEventListener('pointerleave', onPointerLeave)
+  watchPostEffect(onCleanUp => {
```

Comment:
Could you maybe explain to me, why this watcher is here now? Is there a need to add the eventlisteners again and again? 🤔

Reply:
this effect will just run once only the canvas is mounted and it is unmounted, that's similar to onMounted and onUnmounted, there are no one will trigger the canvas's effect,

Reply:
The code also runs, when making changes to the renderer (like toggling `stencil` for example), which is not necessary.

Reply:
but the previous one has broken in some cases, maybe just using onMounted was ok

Reply:
test it on `multiple` route